apps/framework-cli/src/framework/aggregations.ts

```diff
@@ -0,0 +1,164 @@
+import {
+  createClient,
+  ClickHouseClient,
+} from "npm:@clickhouse/client-web@1.0.1";
+import { watch } from "npm:chokidar@3.6.0";
+
+interface ShowTablesResponse {
+  meta: { name: string; type: string }[];
+  data: { name: string }[];
+  rows: number;
+}
+
+const cwd = Deno.args[0] || Deno.cwd();
+const AGGREGATIONS_DIR_PATH = `${cwd}/app/aggregations`;
+const AGGREGATIONS_FILE = "*.ts";
+
+const CLICKHOUSE_DB =
+  Deno.env.get("MOOSE_CLICKHOUSE_CONFIG__DB_NAME") || "local";
+const CLICKHOUSE_HOST =
+  Deno.env.get("MOOSE_CLICKHOUSE_CONFIG__HOST") || "localhost";
+const CLICKHOUSE_PORT =
+  Deno.env.get("MOOSE_CLICKHOUSE_CONFIG__HOST_PORT") || "18123";
+const CLICKHOUSE_USERNAME =
+  Deno.env.get("MOOSE_CLICKHOUSE_CONFIG__USER") || "panda";
+const CLICKHOUSE_PASSWORD =
+  Deno.env.get("MOOSE_CLICKHOUSE_CONFIG__PASSWORD") || "pandapass";
+const CLICKHOUSE_USE_SSL =
+  Deno.env.get("MOOSE_CLICKHOUSE_CONFIG__USE_SSL") || "false";
+
+const getClickhouseClient = () => {
+  const protocol =
+    CLICKHOUSE_USE_SSL.toLowerCase() === "true" ? "https" : "http";
+  console.log(
+    `Connecting to Clickhouse at ${protocol}://${CLICKHOUSE_HOST}:${CLICKHOUSE_PORT}`,
+  );
+  return createClient({
+    url: `${protocol}://${CLICKHOUSE_HOST}:${CLICKHOUSE_PORT}`,
+    username: CLICKHOUSE_USERNAME,
+    password: CLICKHOUSE_PASSWORD,
+    database: CLICKHOUSE_DB,
+  });
+};
+
+const waitForClickhouse = (chClient: ClickHouseClient) => {
```

Comment:
Aggregations is basically materialized views that take data from a source table, aggregates stuff, and puts the result in a target table. Creating a materialized view depends on the source table to exist, so this aggregations service need to wait on something.

I tried ping & show tables, but they aren't 100% reliable. Looks like we already have a ticket for it: https://linear.app/514/issue/514-750/moose-started-faster-than-clickhouse-and-start-up-failed. FYI @phiSgr , lmk if I can help with anything.